### DISCUSSION TITLE
Problems with native MACOS IKEv2: no matching peer config found

### DISCUSSION THREAD

**Discussion by misch42** [strongswan/strongswan#2367] [General]
Hi,

we set up a VPN server with mutual auth by certificates. Works for Linux clients with network manager strongswan plugin or with swanctl CLI.

Now one of our colleges wanted to set up a VPN as a  native MACOS IKEv2 connection. But the connection cannot be established

The log of the VPN server tells me:
   14[CFG] looking for peer configs matching x:x:x:x::1:163[CN=vpn01.example.de]...y:y:y:y:36:2296:5734:2707[O=EXAMPLE.DE, CN=mac_user]
14[CFG] no matching peer config found

The log for the connection from a linus clients looks like:
11[CFG] looking for peer configs matching x:x:x:x::1:163[%any]...y:y:y:y:617d:db61:76b3:1509[O=EXAMPLE.DE, CN=linux_user]
11[CFG] selected peer config 'rw-cert'
11[CFG]   using certificate "O=EXAMPLE.DE, CN=linux_user"
(...)

Why does the strongswan server not find the correct connection?

The connection is defined like:
rw-cert: IKEv2, no reauthentication, rekeying every 14400s
  local:  x:x:x:x::1:163
  remote: %any
  local public key authentication:
    id: CN=vpn01.example.de
    certs: CN=vpn01.example.de
  remote public key authentication:
    id: O=EXAMPLE.DE, CN=*
  net: TUNNEL, rekeying every 3600s
    local:  192.0.2.0/24
    remote: dynamic

**Comment by tobiasbrunner** [comment]
Please see [this FAQ entry](https://docs.strongswan.org/docs/5.9/support/faq.html#_no_matching_peer_config_found) (in particular the last paragraph, most likely `CN=vpn01.example.de` is not actually a DN).

**Comment by misch42** [comment]
Well, that is what my certificate states:

(...)
        Issuer: C = US, O = Let's Encrypt, CN = R3
        Validity
            Not Before: Jun  5 22:01:54 2024 GMT
            Not After : Sep  3 22:01:53 2024 GMT
        Subject: CN = vpn01.example.de
(...)
            X509v3 Subject Alternative Name: 
                DNS:vpn01.example.de
(...)

**Reply by tobiasbrunner** [reply]
It's not a problem with the certificate or the DN matching. The problem is most likely that the client does not send a DN as identity (it might look like one in the log message, but the type is probably something else, increase the log level to see the details). Apple clients always had issues with that, see e.g. the [known issues here](https://docs.strongswan.org/docs/5.9/interop/appleIkev2Profile.html#_known_issues). So just configure the FQDN as server/remote identity on the client.

**Comment by misch42** [comment]
I also changed rdn_matching = relaxed, but no change in behavior.

**Comment by misch42** [comment]
I also added a config with the hostname as local ID of the VPN Server. But still no peer config found.

config:
rw-p: IKEv2, no reauthentication, rekeying every 14400s
  local:  194.126.xxx.xxx
  local:  x:x:x:x::1:163
  remote: %any
  local public key authentication:
    id: vpn01.example.de
    certs: CN=vpn01.example.de
  remote public key authentication:
    id: O=EXAMPLE.DE, CN=p
  net: TUNNEL, rekeying every 3600s
    local:  10.4.4.0/24
    remote: dynamic

Log:
15[IKE] y.y.248.156 is initiating an IKE_SA
15[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/ECP_256
15[IKE] remote host is behind NAT
15[IKE] sending cert request for "C=US, O=Internet Security Research Group, CN=ISRG Root X1"
15[IKE] sending cert request for "O=EXAMPLE.DE, CN=Certificate Authority"
15[IKE] sending cert request for "C=US, O=Let's Encrypt, CN=R3"
15[ENC] generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) N(CHDLESS_SUP) N(MULT_AUTH) ]
15[NET] sending packet: from x.x.158.163[500] to y.y.248.156[62827] (329 bytes)
12[NET] received packet: from y.y.248.156[49346] to x.x.158.163[4500] (400 bytes)
12[ENC] unknown attribute type INTERNAL_DNS_DOMAIN
12[ENC] parsed IKE_AUTH request 1 [ IDi N(INIT_CONTACT) IDr CPRQ(ADDR MASK DHCP DNS ADDR6 DHCP6 DNS6 DOMAIN) N(ESP_TFC_PAD_N) N(NON_FIRST_FRAG) SA TSi TSr N(MOBIKE_SUP) N(EAP_ONLY) ]
12[CFG] looking for peer configs matching x.x.158.163[vpn01.example.de]...y.y.248.156[O=EXAMPLE.DE, CN=p]
12[CFG] no matching peer config found
12[IKE] peer supports MOBIKE
12[IKE] received ESP_TFC_PADDING_NOT_SUPPORTED, not using ESPv3 TFC padding
12[ENC] generating IKE_AUTH response 1 [ N(AUTH_FAILED) ]

**Reply by tobiasbrunner** [reply]
And the identity the client sends as its own, `O=EXAMPLE.DE, CN=p`, is actually a DN? Why don't you increase the log level and check?
